algolia/transport/transport.go

```diff
@@ -302,10 +304,14 @@ func buildRequest(
 }
 
 func unmarshalTo(r io.ReadCloser, v interface{}) error {
-	err := json.NewDecoder(r).Decode(&v)
+	body, err := ioutil.ReadAll(r)
 	errClose := r.Close()
 	if err != nil {
-		return fmt.Errorf("cannot deserialize response's body: %v", err)
+		return fmt.Errorf("cannot read body: %v", err)
+	}
+	err = json.NewDecoder(bytes.NewReader(body)).Decode(&v)
+	if err != nil {
+		return fmt.Errorf("cannot deserialize response's body: `%s` %v", string(body), err)
```

Comment:
```suggestion
		return fmt.Errorf("cannot deserialize response's body: %s %v", string(body), err)
```

Not sure the backtick are that useful. If you see a value you could use `%q`.

Reply:
The goal is to identify where the body starts and ends. I'll replace it with %q thanks!

Reply:
In that case I would probably swap the two `Errorf(body: %v: %s, err, string(body))`. The error is most likely shorter than the entire payload. It would be easier to understand the error and look trough the body after. Otherwise the error message is buried in large blob of text.